src/annotator/plugin/bucket-bar.js

```diff
@@ -69,133 +97,181 @@ export default class BucketBar extends Delegator {
+          scrollToClosest(bucket.anchors, 'down');
+        } else if (this.isUpper(index)) {
+          scrollToClosest(bucket.anchors, 'up');
+        } else {
+          const annotations = bucket.anchors.map(anchor => anchor.annotation);
+          this.annotator.selectAnnotations(
+            annotations,
+            event.ctrlKey || event.metaKey
           );
-        })
-        .on('click', event => {
-          const bucket = this.tabs.index(event.currentTarget);
-          event.stopPropagation();
-
-          // If it's the upper tab, scroll to next anchor above
-          if (this.isUpper(bucket)) {
-            scrollToClosest(this.buckets[bucket], 'up');
-            // If it's the lower tab, scroll to next anchor below
-          } else if (this.isLower(bucket)) {
-            scrollToClosest(this.buckets[bucket], 'down');
-          } else {
-            const annotations = this.buckets[bucket].map(
-              anchor => anchor.annotation
-            );
-            this.annotator.selectAnnotations(
-              annotations,
-              event.ctrlKey || event.metaKey
-            );
-          }
-        });
+        }
+      });
+
+      this.element.append(tabEl);
     });
 
     this._buildTabs();
   }
 
   _buildTabs() {
-    this.tabs.each((index, el) => {
-      let bucketSize;
-      el = $(el);
-      const bucket = this.buckets[index];
-      const bucketLength = bucket?.length;
-
-      const title = (() => {
-        if (bucketLength !== 1) {
-          return `Show ${bucketLength} annotations`;
-        } else if (bucketLength > 0) {
-          return 'Show one annotation';
+    this.tabs.forEach((tabEl, index) => {
+      const anchorCount = this.buckets[index].anchors.length;
+      // Positioning logic currently _relies_ on their being interstitial
+      // buckets that have no anchors but do have positions. Positioning
+      // is averaged between this bucket's position and the _next_ bucket's
+      // position. For now. TODO: Fix this
+      const pos =
+        (this.buckets[index].position + this.buckets[index + 1]?.position) / 2;
+      let bucketHeight = BUCKET_SIZE;
+
+      tabEl.className = 'annotator-bucket-indicator';
+      tabEl.style.top = `${pos}px`;
+      tabEl.style.display = '';
+
+      if (anchorCount) {
+        tabEl.innerHTML = `<div class="label">${this.buckets[index].anchors.length}</div>`;
+        if (anchorCount === 1) {
+          tabEl.setAttribute('title', 'Show one annotation');
+        } else {
+          tabEl.setAttribute('title', `Show ${anchorCount} annotations`);
         }
-        return '';
-      })();
-
-      el.attr('title', title);
-      el.toggleClass('upper', this.isUpper(index));
-      el.toggleClass('lower', this.isLower(index));
-
-      if (this.isUpper(index) || this.isLower(index)) {
-        bucketSize = BUCKET_NAV_SIZE;
       } else {
-        bucketSize = BUCKET_SIZE;
+        tabEl.style.display = 'none';
```

Comment:
I'd really like to refactor this...soon. Goofballs to have all these extra elements that are hidden.